windows/src/engine/kmtip/keyman32interface.cpp

```diff
@@ -1,24 +1,42 @@
 #include "pch.h"
 #include "kmtip.h"
 
-typedef BOOL (WINAPI *PKEYMAN_WRITEDEBUGEVENT)(char *file, int line, PWCHAR msg);
-
-//TODO: LoadLibrary on startup, FreeLibrary on shutdown
-HMODULE Keyman32Interface::GetHandle() {
-#ifdef _WIN64
-  return GetModuleHandle("keyman64.dll");
-#else
-  return GetModuleHandle("keyman32.dll");
-#endif
-}
+extern "C" {
+  BOOL WINAPI Keyman_WriteDebugEvent(char *file, int line, PWCHAR msg);
+
+  BOOL WINAPI TIPActivateEx(BOOL fActivate);
+  BOOL WINAPI TIPActivateKeyboard(GUID *profile);
+
+  BOOL WINAPI TIPProcessKey(WPARAM wParam, LPARAM lParam, PKEYMANPROCESSOUTPUTFUNC outfunc,
+    PKEYMANGETCONTEXTFUNC ctfunc, BOOL Updateable, BOOL Preserved);
+
+  BOOL WINAPI GetKeyboardPreservedKeys(PreservedKey **pPreservedKeys, size_t *cPreservedKeys);
+
+  BOOL WINAPI TIPIsKeymanRunning();
+};
 
 void Keyman32Interface::WriteDebugEvent(char *file, int line, PWCHAR msg) {
-  HMODULE hKeyman = GetHandle();
-  if (hKeyman) {
-    PKEYMAN_WRITEDEBUGEVENT pWriteDebugEvent = (PKEYMAN_WRITEDEBUGEVENT)GetProcAddress(hKeyman, "Keyman_WriteDebugEvent");
-    if (pWriteDebugEvent) {
-      (*pWriteDebugEvent)(file, line, msg);
-    }
-    // Don't DebugLastError here because it depends on WriteDebugEvent...
-  }
+  ::Keyman_WriteDebugEvent(file, line, msg);
+  // Don't DebugLastError here because it depends on WriteDebugEvent...
+}
```

Comment:
I've noticed this pattern shift a few times during the review.  I assume it accomplishes the same goal, just in a more polished way?